app/scripts/multichain-router-messenger.ts

```diff
@@ -0,0 +1,56 @@
+import { Messenger } from '@metamask/base-controller';
+import {
+  AccountsControllerAccountAddedEvent,
+  AccountsControllerAccountRemovedEvent,
+  AccountsControllerListMultichainAccountsAction,
+  AccountsControllerAccountBalancesUpdatesEvent,
+} from '@metamask/accounts-controller';
+import { HandleSnapRequest } from '@metamask/snaps-controllers';
+import {
+  MultichainAssetsControllerGetStateAction,
+  MultichainAssetsControllerStateChangeEvent,
+} from '@metamask/assets-controllers';
+import { KeyringControllerGetStateAction } from '@metamask/keyring-controller';
+
+type Actions =
+  | AccountsControllerListMultichainAccountsAction
+  | HandleSnapRequest
+  | MultichainAssetsControllerGetStateAction
+  | KeyringControllerGetStateAction;
+
+type Events =
+  | AccountsControllerAccountAddedEvent
+  | AccountsControllerAccountRemovedEvent
+  | AccountsControllerAccountBalancesUpdatesEvent
+  | MultichainAssetsControllerStateChangeEvent;
+
+export type MultichainBalancesControllerMessenger = ReturnType<
+  typeof getMultichainBalancesControllerMessenger
+>;
+
+/**
+ * Get a restricted messenger for the Multichain Balances controller. This is scoped to the
+ * actions and events that the Multichain Balances controller is allowed to handle.
+ *
+ * @param messenger - The controller messenger to restrict.
+ * @returns The restricted controller messenger.
+ */
+export function getMultichainBalancesControllerMessenger(
```

Comment:
🤔 copy pasta?

Reply:
yes

Reply:
ahh... i copied this the mutlichain balances messenger file but editted the wrong copy

Reply:
oh no. this is just a extra copied file that isn't needed